NodeJS/src/AgentClient.ts

```diff
@@ -0,0 +1,413 @@
+                            clearInterval(intervalId);
+                            reject(Error('station creation failed'))
+                        }
+                    }
+                } catch (e) {
+                    if (e.response.status === 404) {
+                        process.stdout.write('.');
+                        counter++;
+                        if (counter > MAX_RETRY) {
+                            console.log();
+                            clearInterval(intervalId);
+                            reject(Error('station creation failed'))
+                        }
+                    } else {
+                        console.log(e);
+                        throw e
+                    }
+                }
+            }, INTERVAL_IN_MILLIS);
+            return intervalId
+        })
+    }
+
+    async waitForStationDeletion(subAccountId: any, agent_username: any) {
+        return new Promise((resolve, reject) => {
+            process.stdout.write("Deleting station.");
+            let counter = 0;
+            const intervalId = setInterval(async () => {
+                try {
+                    let station = await this.getStationOnly(subAccountId, agent_username);
+                    if (station) {
+                        process.stdout.write('.');
+                        counter++;
+                        if (counter > MAX_RETRY) {
+                            console.log();
+                            clearInterval(intervalId);
+                            reject(Error('station deletion failed'))
+                        }
+                    } else {
+                        console.log('station deleted');
+                        clearInterval(intervalId);
+                        resolve()
+                    }
+                } catch (e) {
+                    if (e.response.status === 404) {
+                        console.log('station deleted');
+                        clearInterval(intervalId);
+                        resolve()
+                    } else {
+                        console.log(e);
+                        throw e
+                    }
+                }
+            }, INTERVAL_IN_MILLIS);
+            return intervalId
+        })
+    }
+
+    async getStationOnly(subAccountId: string, agent_username: string) {
+        return this.session.get('/spokenAbc/stations?clientId=' + subAccountId)
+            .then((result: { data: any[] }) => {
+                return result.data.find(element => element.username === agent_username);
+            })
+    }
+
+    async deleteAgent(agentUsername: string) {
+        await this.session.login();
+
+        let subAccountId = await this.getSubAccountId();
+
+        let station = await this.getStationOnly(subAccountId, agentUsername);
+        // station might have been deleted before, so station might be undefined
+        if (station) {
+            await this.deleteStationOnly(station.id);
```

Comment:
should these two lines be put inside a try catch block?

Reply:
If there is an exception the application would not swallow it . It will just thrown to the top. We dont need to worry here.  The reason we added a catch on other areas is we are doing some additional process on the exception.